src/main/kotlin/hwr/oop/tnp/Monster.kt

```diff
@@ -0,0 +1,42 @@
+package hwr.oop.tnp
+
+import kotlin.math.max
+
+class Monster(
+    private val name: String,
+    private val stats: BattleStats,
+    private val type: Type,
+    private val attacks: List<Attack>
+) {
+    private var hp: Int = stats.hp
+
+    fun getHp(): Int {
+        return this.hp
+    }
```

Comment:
Dont declare `hp` here but use `this.stats.hp` or something similar that works instead
```suggestion
    fun getHp(): Int {
        return this.stats.hp
    }
```